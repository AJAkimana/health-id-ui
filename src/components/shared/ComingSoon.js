import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import {
  Paper,
  withStyles,
  Typography,
  Grid,
  Card,
  CardMedia
} from '@material-ui/core';
import withAuth from '../withAuth';
import { comingSoonStyles } from '../../assets/styles/comingsoon';
import ComingSoonImage from '../../assets/images/comingsoon.gif';

const ComingSoon = (props) => {
  const { classes } = props;
  return (
    <Fragment>
      <Paper className={classes.paper}>
        <Grid container spacing={24} className={classes.containerGrid}>
          <Grid container item xs={12}>
            <Grid container item xs={8} spacing={24}>
              <Grid item xs={12}>
                <Typography variant="h5" className={classes.arrowButtonLabel}>
                  Coming soon
                </Typography>
              </Grid>
            </Grid>
            <Grid item xs={4}>
              <Card elevation={0} className={classes.card}>
                <CardMedia className={classes.media} image={ComingSoonImage} title="Coming soon" />
              </Card>
            </Grid>
          </Grid>
        </Grid>
      </Paper>
    </Fragment>
  );
};
ComingSoon.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string)
};

ComingSoon.defaultProps = {
  classes: {}
};

export default withAuth(withStyles(comingSoonStyles)(ComingSoon));
