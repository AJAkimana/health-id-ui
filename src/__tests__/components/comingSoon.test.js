import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import MockComponent from '../../../__mocks__/mockComponent';
import toJson from 'enzyme-to-json';
import ComingSoon from '../../components/shared/ComingSoon';

jest.mock('../../components/shared/ComingSoon', () => MockComponent);

const props = {
  variables: {},
  classes: {},
  session: {
    me: {
      role: {
        name: 'Master Admin'
      },
      activeOutlet: {
        outletpreference: {
          outletTimezone: {
            name: "Africa/Nairobi"
          }
        }
      }
    }
  }
};

describe('ComingSoon', () => {
  it('renders without crashing', async() => {
    const wrapper = mount(
      <MockedProvider addTypeName={false}>
        <BrowserRouter>
          <ComingSoon {...props} />
        </BrowserRouter>
      </MockedProvider>
    );
  
    expect(wrapper).toHaveLength(1);
  });

  it('should render the ComingSoon component properly', () => {
    const setupComingSoon = () => {
      const wrapper = shallow(<ComingSoon {...props} />)
      return wrapper
    }

    let wrapper;
    wrapper = setupComingSoon();
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
