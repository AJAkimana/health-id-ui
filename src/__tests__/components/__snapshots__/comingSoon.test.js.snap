// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComingSoon should render the ComingSoon component properly 1`] = `<div />`;
